### PR TITLE
feat: Gather per-container network statistics

### PR DESCRIPTION
**Pull request by sanxiyn**

### CODE REVIEW

File: src/ai/backend/agent/docker/intrinsic.py [hunk 1]
```diff
@@ -597,9 +598,29 @@ async def sysfs_impl(container_id):
                     e,
                 )
                 return None
+            async with closing_async(Docker()) as docker:
+                container = DockerContainer(docker, id=container_id)
+                data = await container.show()
+                sandbox_key = data["NetworkSettings"]["SandboxKey"]
```
Comment (achimnol) [inline src/ai/backend/agent/docker/intrinsic.py:625]: How about storing this sandbox key in `kernel_registry`?

File: src/ai/backend/agent/docker/intrinsic.py [hunk 2]
```diff
@@ -597,9 +598,29 @@ async def sysfs_impl(container_id):
...
+            net_rx_bytes = 0
+            net_tx_bytes = 0
+            with nsenter(sandbox_key):
+                nstat = psutil.net_io_counters(pernic=True)
```
Comment (achimnol) [inline src/ai/backend/agent/docker/intrinsic.py]: Would this affect existing socket connections though it should take a very short amount of time?

Comment (sanxiyn) [reply src/ai/backend/agent/docker/intrinsic.py]: Oops, you are right. This should be executed in the new thread.

File: src/ai/backend/agent/docker/intrinsic.py (hunk 1 above)
Comment (sanxiyn) [reply src/ai/backend/agent/docker/intrinsic.py:625]: Sure.

Comment (sanxiyn) [reply src/ai/backend/agent/docker/intrinsic.py:625]: I created a new issue for this: #1196.

File: src/ai/backend/agent/docker/intrinsic.py (hunk 2 above)
Comment (sanxiyn) [reply src/ai/backend/agent/docker/intrinsic.py]: Rather, this should be executed in the new process... See my new commit and comment there for details.

**Review by achimnol — APPROVED** [review]

**Pull request merged**